Dockerfile

```diff
@@ -3,24 +3,16 @@ ARG DEBIAN_FRONTEND=noninteractive
 
 # Install dependencies
 RUN apt-get update && \
-    apt-get install -y git clang clang-format libboost-all-dev libeigen3-dev rapidjson-dev build-essential gcc make cmake libssl-dev libncurses5-dev libncursesw5-dev
-
-# Install Paho
-WORKDIR /pahoc
-RUN git clone --branch v1.2.1 https://github.com/eclipse/paho.mqtt.c.git
-WORKDIR /pahoc/paho.mqtt.c
-RUN make
-RUN make install
-
+    apt-get install -y git clang clang-format libboost-all-dev libeigen3-dev rapidjson-dev build-essential gcc make cmake libssl-dev libncurses5-dev libncursesw5-dev sudo
+    
 # Install Paho C++
-WORKDIR /pahocpp
-RUN git clone --branch v1.0.0 https://github.com/eclipse/paho.mqtt.cpp
-WORKDIR /pahocpp/paho.mqtt.cpp
-RUN mkdir build
-WORKDIR /pahocpp/paho.mqtt.cpp/build
-RUN cmake -DPAHO_BUILD_DOCUMENTATION=FALSE -DPAHO_BUILD_SAMPLES=FALSE -DPAHO_MQTT_C_PATH=/pahoc/paho.mqtt.c ..
-RUN make
-RUN make install
+WORKDIR /home
+RUN git clone --recurse-submodules https://github.com/eclipse/paho.mqtt.cpp.git
+WORKDIR /home/paho.mqtt.cpp
+RUN git checkout ef021805c7aca42d72194d85bb2bdccdfe963e28
+RUN cmake -Bbuild -H. -DPAHO_WITH_MQTT_C=ON -DPAHO_BUILD_STATIC=ON -DPAHO_BUILD_DOCUMENTATION=OFF -DPAHO_BUILD_SAMPLES=ON
+RUN sudo cmake --build build/ --target install
+RUN sudo ldconfig
```

Comment:
what does this do?

Reply:
from what I can tell it's something to do with updating symlinks and keeping track of versions, so idk if its necessary. do you think its worth keeping?

Reply:
where did you get it from?

if it's in the official install guide (and you trust the guide lol) then it should stay

Reply:
yeah it was just part of the install guide so probs best to keep it